processor/src/main/java/org/mapstruct/ap/internal/model/source/SourceMethod.java

```diff
@@ -590,6 +592,11 @@ public boolean isBeforeMappingMethod() {
         return Executables.isBeforeMappingMethod( getExecutable() );
     }
 
+    public boolean isAbstract() {
```

Comment:
This one needs to represent whether an implementation will be created for this method, right? 

Why don't we do `return overridesMethod() || !isDefault()`? Or maybe we need to do something else, as this will pick up Java 8 default methods. I think that we need to consider the Java 8 default methods the same way as we consider non abstract methods.